src/RequestHandler.cpp

```diff
@@ -28,24 +27,29 @@ void RequestHandler::checkAndRun() {
 /// \param UserArguments
 /// \param TerminateAtEndOfTopic terminate at end of topic. For unit tests.
 void RequestHandler::checkConsumerModeArguments(bool TerminateAtEndOfTopic) {
+  if (!UserArguments.JSONPath.empty()) {
+    throw ArgumentException("Cannot run consumer mode with -f specified. Did "
+                            "you want to use -P mode?");
+  }
   if (UserArguments.GoBack == -2 && UserArguments.OffsetToStart == -2 &&
-      UserArguments.Name.empty()) {
+      UserArguments.TopicName.empty()) {
     throw ArgumentException("Please specify topic!");
   }
   if (UserArguments.GoBack == -2 && UserArguments.OffsetToStart == -2) {
-    printEntireTopic(UserArguments.Name, TerminateAtEndOfTopic);
+    printEntireTopic(UserArguments.TopicName, TerminateAtEndOfTopic);
   } else {
-    checkIfTopicEmpty(UserArguments.Name);
+    checkIfTopicEmpty(UserArguments.TopicName);
     if (UserArguments.GoBack > -2 && UserArguments.OffsetToStart > -2) {
-      subscribeAndConsume(UserArguments.Name, UserArguments.GoBack,
+      subscribeAndConsume(UserArguments.TopicName, UserArguments.GoBack,
                           UserArguments.PartitionToConsume,
                           UserArguments.OffsetToStart);
     } else {
       if (UserArguments.OffsetToStart > -2) {
-        subscribeAndConsume(UserArguments.Name, UserArguments.OffsetToStart);
+        subscribeAndConsume(UserArguments.TopicName,
```

Comment:
codecov says this line is not covered, it this a scenario which should have a unit test?

Reply:
The way this scenario is tested changed after we introduced that infinite loop. Testing starts with method that is called by this `subscribeAndConsume` which has a boolean preventing the innfinite loop from running. Starting testing here makes tests hang.